commit cd7f4d153ccccf601751e9fa82424412f6ecfc96 (HEAD)
Author: Eric Blake <eblake@redhat.com>
Date:   Tue Jun 1 08:10:51 2021 -0500

    tests: Fix 198.sysval

    In my attempt to avoid test failures on Haiku, I caused test failures
    on platforms where sh is noisy when reporting a killed sub-process.

    * doc/m4.texi (Sysval): Avoid stderr noise during test.
    Fixes: 17011ea76a (tests: Skip signal detection on Haiku)
    Fixes: https://lists.gnu.org/archive/html/bug-m4/2021-05/msg00029.html

diff -upr m4-1.4.19.orig/checks/198.sysval m4-1.4.19/checks/198.sysval
--- m4-1.4.19.orig/checks/198.sysval	2022-01-30 12:07:12.283721677 +0100
+++ m4-1.4.19/checks/198.sysval	2022-01-30 12:08:03.733939699 +0100
@@ -14,7 +14,7 @@ ifdef(`__unix__', ,
 ')m4exit(`77')')dnl
 changequote(`[', `]')
 dnl @result{}
-syscmd([/bin/sh -c 'kill -9 $$'; st=$?; test $st = 137 || test $st = 265])
+syscmd([@{ /bin/sh -c 'kill -9 $$'; @} 2>/dev/null; st=$?; test $st = 137 || test $st = 265])
 dnl @result{}
 ifelse(sysval, [0], , [errprint([ skipping: shell does not send signal 9
 ])m4exit([77])])dnl
